### PR TITLE
squid:S1193 - Exception types should not be tested using "instanceof" in catch blocks

### PR DESCRIPTION
**Pull request by georgekankava**
This pull request is focused on resolving occurrences of Sonar rule squid:S1193 - Exception types should not be tested using "instanceof" in catch blocks.
You can find more information about the issue here:
https://dev.eclipse.org/sonar/coding_rules#q=squid%3AS1193
Please let me know if you have any questions.
George Kankava
Unwatch this pull request
Learn more

### CODE REVIEW

**Comment by hazendaz** [comment]
Pretty sure when rethrowinf thread death the casting isn't needed.  Can you verify that?


Sent from Outlook Mobile






On Tue, Feb 2, 2016 at 9:13 AM -0800, "George Kankava" <notifications@github.com> wrote:





This pull request is focused on resolving occurrences of Sonar rule squid:S1193 - Exception types should not be tested using "instanceof" in catch blocks.
You can find more information about the issue here:
https://dev.eclipse.org/sonar/coding_rules#q=squid%3AS1193
Please let me know if you have any questions.
George Kankava
Unwatch this pull request
Learn more

You can view, comment on, or merge this pull request online at:

  https://github.com/psi-probe/psi-probe/pull/638

-- Commit Summary --

  * squid:S1193 - Exception types should not be tested using "instanceof" in catch blocks

-- File Changes --

    M core/src/main/java/psiprobe/beans/LogResolverBean.java (56)
    M core/src/main/java/psiprobe/beans/ResourceResolverBean.java (9)

-- Patch Links --

https://github.com/psi-probe/psi-probe/pull/638.patch
https://github.com/psi-probe/psi-probe/pull/638.diff

---
Reply to this email directly or view it on GitHub:
https://github.com/psi-probe/psi-probe/pull/638

**Comment by georgekankava** [comment]
Yes I have verified that before committing and it catches ThreadDeath error
without any casting needs just like any other exception.

Thanks,
George

File: core/src/main/java/psiprobe/beans/LogResolverBean.java [hunk 1]
```diff
@@ -325,11 +325,10 @@ private void interrogateClassLoader(ClassLoader cl, Application application,
       Jdk14ManagerAccessor jdk14accessor = new Jdk14ManagerAccessor(cl);
       jdk14accessor.setApplication(application);
       appenders.addAll(jdk14accessor.getHandlers());
-    } catch (Throwable t) {
+    } catch (ThreadDeath t) {
       // make sure we always re-throw ThreadDeath
-      if (t instanceof ThreadDeath) {
-        throw (ThreadDeath) t;
-      }
+      throw (ThreadDeath) t;
```
Comment (hazendaz) [inline core/src/main/java/psiprobe/beans/LogResolverBean.java]: @georgekankava This is the casting I was mentioning.  As you have this catching threadDeath now, the casting is unnecessary.  Can you fix this and squash the commit so I can merge it?  There are a number of these so please review the overall change.  Thanks.

Comment (georgekankava) [inline core/src/main/java/psiprobe/beans/LogResolverBean.java]: Sure I will do that. Thank you.

**Comment by hazendaz** [comment]
@georgekankava can you squash the commits for this?

Steps including rebase to master...

git pull --rebase upstream master
git rebase -i HEAD~4
follow directions for squashing
git push origin +staging/exception-types-should-not-be-tested-using-instanceof-in-catch-blocks-fix-1

Then I'll merge this right in.  Thanks.

**Comment by hazendaz** [comment]
I'm going to take care of this.  This will be on master based on your actual PR shortly.

**Comment by hazendaz** [comment]
Applied on #652

**Pull request closed without merging**